src/models/FilterString.js

```diff
@@ -4,22 +4,39 @@
  * @param {String} query The search query
  */
 
-import Filter from '@/models/FilterBase';
+import FilterItems from '@/models/FilterItems';
 import * as precisions from './Precisions';
 import * as contexts from './Contexts';
 
-export default class FilterString extends Filter {
+class StringItem {
+  constructor({
+    uid = '',
+  }) {
+    this.uid = uid;
+    this.isValid = typeof this.uid === 'string' && this.uid.trim().length > 0;
+  }
+}
+
+export default class FilterString extends FilterItems {
   constructor(args) {
     super(args);
-    this.q = String(args.q || args.query);
     this.precision = precisions[(args.precision || 'exact').toUpperCase()];
     this.distance = Math.max(0, Math.min(parseInt(args.distance || 0, 10), 10));
   }
 
+  setItems(items = []) {
+    this.items = items.map((uid) => {
+      const item = new StringItem({ uid });
+      item.checked = true;
```

Comment:
What is the `checked` flag for? Looks like it is always true. Do we need it?

Reply:
true, I'll remove this, it was related to the previous implementation, using checkboxes

Reply:
Yeah it has been redone with lists which makes it a bit cleaner.